### PR TITLE
Update v13 Migration Guide

### PR DESCRIPTION
**Pull request by tobias-tengler**

### CODE REVIEW

**Comment by tobias-tengler** [comment]
@michaelstaib I noticed two new breaking changes while scanning your latest DataLoader related commit, so I've updated the migration guide. 

Do you know of any other breaking changes? I only know about the TypeInterceptor changes. We should probably add new breaking changes to this document immediately, so we can provide a better update experience once we release v13.

**Comment by michaelstaib** [comment]
Well spotted... there are more :) I will leave this PR open and add to it.

File: website/src/docs/hotchocolate/v13/migrating/migrate-from-12-to-13.md [hunk 1]
```diff
@@ -1,3 +1,135 @@
...
+  # The authorization rule is now evaluated here, since the `user` field
+  # returns the `User` type, which is annotated with @authorize
+  user {
+    field1
+    field2
+  }
+}
+```
+
+This behavior shouldn't be a breaking change regarding functionality, but the authorization error will now be raised at the field returning the type instead of the field beneath the type, which might break your test assertions.
```
Comment (tobias-tengler) [inline website/src/docs/hotchocolate/v13/migrating/migrate-from-12-to-13.md:98]: @michaelstaib I haven't played around with the new `@authorize` behavior so I'm not sure if this is correct. Could you look it over please?

**Comment by michaelstaib** [comment]
I have to write more on authorization ... but I added a small section on how to secure the root types.

**Pull request merged**